src/world.rs

```diff
@@ -59,14 +59,14 @@ struct Appendix {
 }
 
 fn find_next(gens: &[Generation], lowest_free_index: usize) -> Entity {
-    if let Some((id, gen)) = gens.iter().enumerate().skip(lowest_free_index).find(|&(_, g)| *g <= 0) {
-        return Entity(id as Index, 1 - gen);
+    if let Some((id, gen)) = gens.iter().enumerate().skip(lowest_free_index).find(|&(_, g)| g.is_dead()) {
+        return Entity(id as Index, Generation(1 - gen.0));
     }
 
     if lowest_free_index > gens.len() {
-        Entity(lowest_free_index as Index, 1)
+        Entity(lowest_free_index as Index, Generation(1))
```

Comment:
Please refactor this to only return the index id from the condition, and then return the `Entity` in a separate line